arangod/Aql/ExecutionNode/SingletonNode.cpp

```diff
@@ -33,12 +33,24 @@
 using namespace arangodb;
 using namespace arangodb::aql;
 
+SingletonNode::SingletonNode(ExecutionPlan* plan, ExecutionNodeId id,
+                             BindParameterVariableMapping bindParameterOutVars)
+    : ExecutionNode(plan, id),
+      _bindParameterOutVars(std::move(bindParameterOutVars)) {}
+
 SingletonNode::SingletonNode(ExecutionPlan* plan, ExecutionNodeId id)
     : ExecutionNode(plan, id) {}
 
 SingletonNode::SingletonNode(ExecutionPlan* plan,
                              arangodb::velocypack::Slice base)
-    : ExecutionNode(plan, base) {}
+    : ExecutionNode(plan, base) {
+  if (auto bindVars = base.get("bindParameterVariables"); !bindVars.isNone()) {
+    for (auto [key, value] : VPackObjectIterator(bindVars)) {
```

Comment:
nit: Iteration order probably does not matter here
```suggestion
    for (auto [key, value] : VPackObjectIterator(bindVars, true)) {
```